pkg/registryclient/client.go

```diff
@@ -191,10 +200,25 @@ func (c *client) Options(ctx context.Context) ([]gcrremote.Option, error) {
 	return opts, nil
 }
 
+// NameOptions returns name.Option config parameters for the client
+func (c *client) NameOptions() []name.Option {
+	nameOpts := []name.Option{}
+
+	if c.allowInsecureRegistry {
+		nameOpts = append(nameOpts, name.Insecure)
+	}
+
+	return nameOpts
+}
+
 // FetchImageDescriptor fetches Descriptor from registry with given imageRef
 // and provides access to metadata about remote artifact.
 func (c *client) FetchImageDescriptor(ctx context.Context, imageRef string) (*gcrremote.Descriptor, error) {
-	parsedRef, err := name.ParseReference(imageRef)
+	nameOption := []name.Option{}
+	if c.allowInsecureRegistry {
+		nameOption = append(nameOption, name.Insecure)
+	}
```

Comment:
you should use the NameOptions function here

Reply:
Fixed